cli/command/context.go

```diff
@@ -0,0 +1,44 @@
+package command
+
+import (
+	"github.com/docker/docker/pkg/contextstore"
+)
+
+// ContextMetadata is a typed representation of what we put in Context metadata
```

Comment:
I'm a little confused about the relationship between what is here and what is in the moby/moby/client stuff e.g. the `Context` defined there but also the `consts` subpackage. They both seem to cover various aspects of the method for a client to connect to an engine.

What is the division of responsibility here? Do we need to "namespace" the metadata keys somehow?

Reply:
The contextstore is completely typeless. It only stores key-value pairs.
ContextMetadata in this package is simply a typed wrapper.

Reply:
In this case though their appeared to be two such typed wrappers covering the same "connect to Docker Engine" configuration case, this one and another one which was `moby/moby/client.Context` in that PR. 

That's what I don't get.

Reply:
Nope, there is a wrapper around context-scope metadata (with only description and defaultStackOrchestrator), and 2 at the endpoint level (1 in cli/context/docker and 1 in cli/context/kubernetes). To better differentiate them, endpoints wrappers are now correctly named as "Endpoint" :)

Reply:
Great, hopefully it'll all become clear when I go through it all again then!